app/src/main/kotlin/com/wire/android/util/ui/WireSessionImageLoader.kt

```diff
@@ -79,14 +85,20 @@ class WireSessionImageLoader(private val coilImageLoader: ImageLoader) {
                     memoryCacheKey = null
                 )
                 .build(),
+            error = errorDrawable?.let { painterResource(id = it) },
             imageLoader = coilImageLoader
         )
 
-        if (painter.state is AsyncImagePainter.State.Error) {
-            val errorCode = ((painter.state as AsyncImagePainter.State.Error).result.throwable as? HttpException)?.response?.code
+        LaunchedEffect(key1 = asset) {
+            networkStateObserver.observeNetworkState().collect { state ->
```

Comment:
This will only react when network state changes, not when `painter.state` changes - are you sure this will work as expected? 🤔

Reply:
`LaunchedEffect(painter.state) { 
	if (painter.state is AsyncImagePainter.State.Error) {
		val isRetryNeeded = ((painter.state as AsyncImagePainter.State.Error).result.throwable as? AssetImageException) ?.isRetryNeeded ?: false
		if (isRetryNeeded) {
			networkStateObserver.observeNetworkState().firstOrNull { it == NetworkState.ConnectedWithInternet }.let {
				retryHash += RETRY_INCREMENT_ATTEMPT_PER_STEP
			}
		}
	}
}`
Updated with this solution